### PR TITLE
Formatting in extensions

### PR DESCRIPTION
**Pull request by willtebbutt**
For reasons that I'm not entirely clear on, I had really poorly formatted basically all of the package extensions. This PR sorts that out. I'll merge when CI passes. Since there are _literally_ no changes to functionality, I won't bother changing the patch version.

edit: this isn't urgent, so I'm going to try and figure out how to sort coverage out before merging this. I imagine we'll be doing more GPU stuff in the comparatively near future, so now is a good time to get it sorted.

### CODE REVIEW

**Comment by willtebbutt** [comment]
@maleadt any idea how I can get make my codecov token available in buildkite? I don't think I had codecov turned on when you installed buildkite in this org, so maybe there's something else that we need to do to enable coverage?

**Comment by maleadt** [comment]
You need to add the token as an encrypted secret to the pipeline: https://github.com/JuliaGPU/buildkite?tab=readme-ov-file#using-secrets

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -13,3 +13,5 @@ steps:
     timeout_in_minutes: 60
     env:
       TEST_GROUP: "gpu"
+env:
+  SECRET_CODECOV_SECRET: "O3JDNB8okS77naB4HEUJkS9jBDuCAAawjKp4wio5T5TAjacTjr+aSzx9Zp0Zp34XywZJjnQTIfBs7ryK46hA5oli20QBTBDBaEaxCDLWnrqpIe0N4KXiRMFthds8U3NT9rnCej6XehdhIU6qtbXjd8I0rAnJYJZQ1ffrlXzfhNX35zuKo2UZ4tl5aMFiMf5bTBLxe9d3F1xUthbLEQmePZYGbHfpJ7K/6op44UF9UL+GIV2MgKS0ZmkiPSBz2ICa/r6ny6iZsKU3ddM38eQfjIE50c42X/3JDjQhP4Pb/OTPbGk6udXISbjrascxgB6OC+oILfm53sME4ARdlZV7MA==;U2FsdGVkX19vGkyy2rbTqUEV8Q1cJK6muQec2lufYO8="
```
Comment (willtebbutt) [inline .buildkite/pipeline.yml]:
```suggestion
env:
  SECRET_CODECOV_TOKEN: "evD6ybRun2WW45mxNN8GcGySZi1/f9lZRlE2sv3tP/Vj8pU7QdGQ6mYrMauhnpU7RiVUBbeypi6YFGqMSbanbpmKZ32JOrlc841KYt4iJnpUsY3RrjXsRN39bJZ8b/qMcFiF5U0JcENb1/nLs8OzeFLZECxNKX0lK2OlrZ7ZcEGfa4xFsBSlnUAxWbUzTrg0xgfCrLIEp2oRN3HzvdV3kgNPyKgqRVOd/qbgLu+RuElkUj7T3AAfPgn6f57qiEIwPQ8gDIxwVfpxGyUWwcGitadfW4hGEHmliLxvPNAxCUz+FIyvTmviOV/5wJYHOYeOeymQO1qFYi8PpOg4odY8YA==;U2FsdGVkX1+W15397ySKjT/XFmT88oi/K542mpVC+4c="
```

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -13,3 +13,5 @@ steps:
     timeout_in_minutes: 60
     env:
       TEST_GROUP: "gpu"
+env:
+  SECRET_CODECOV_TOKEN: "evD6ybRun2WW45mxNN8GcGySZi1/f9lZRlE2sv3tP/Vj8pU7QdGQ6mYrMauhnpU7RiVUBbeypi6YFGqMSbanbpmKZ32JOrlc841KYt4iJnpUsY3RrjXsRN39bJZ8b/qMcFiF5U0JcENb1/nLs8OzeFLZECxNKX0lK2OlrZ7ZcEGfa4xFsBSlnUAxWbUzTrg0xgfCrLIEp2oRN3HzvdV3kgNPyKgqRVOd/qbgLu+RuElkUj7T3AAfPgn6f57qiEIwPQ8gDIxwVfpxGyUWwcGitadfW4hGEHmliLxvPNAxCUz+FIyvTmviOV/5wJYHOYeOeymQO1qFYi8PpOg4odY8YA==;U2FsdGVkX1+W15397ySKjT/XFmT88oi/K542mpVC+4c="
```
Comment (willtebbutt) [inline .buildkite/pipeline.yml]:
```suggestion
env:
  SECRET_CODECOV_TOKEN: "M+fbiuiAVojU6VIquhJ6+oC/EI4hL8Jfubdnc03HV1t0Fpn//CxfnTZi/lT8SoBromLOCIYhQdhCrKla8VgnWVPNmcmrpBkEHtJJ6egmclwwYQGsPAsFrubosSquhK3IGgr2JOxQo4Aa9t5tx1yahEeRxQOKKhgF4pKIscGCz7OYGTIivOMg69AFPKl91i/8rtwVjS4Q2cBj0+reA8jx0DiM70tQe0ceYdXOCczfSkE1iegpSzreYTPoKc6DDt0keTr6FqRlUPKr4TuqA+smYV91MY7OUwi1w7kb0nkzPyCES72T9i1RBAHhC0p28WVRUEkOn7X/V4nhiZOodaStRw==;U2FsdGVkX18+XJdcOzrVVhv3a6+jf7qxRz0TxBPNLyXzoom+BTQu+CKUjld70VdDUGSUseqiNXS9ZdnNIAV53w=="
```

**Comment by willtebbutt** [comment]
Thanks for the pointer @maleadt -- I've updated `.pipeline.yml` but it doesn't appear to be working as expected. Can you see anything that I'm doing obviously wrong?

**Comment by maleadt** [comment]
Our logic to determine if this is a third-party PR seems to fail: https://github.com/JuliaGPU/buildkite/blob/0dbc7b3422d53d770b8e26d79cbada81a0e6d5ce/image/hooks/environment#L5-L8
```
BUILDKITE_PULL_REQUEST_REPO="https://github.com/compintell/Tapir.jl.git"
BUILDKITE_REPO="https://github.com/withbayes/Tapir.jl.git"
```

Was this repository renamed?

**Comment by willtebbutt** [comment]
Ah, yes -- the org was renaming from `withbayes` to `compintell`

**Comment by maleadt** [comment]
I updated the repository, and now it does successfully decrypt the token:

```
2024-09-16 09:49:28 CEST	🔑 Decrypting secrets
2024-09-16 09:49:28 CEST	Found secret for CODECOV_TOKEN
2024-09-16 09:49:28 CEST	# CODECOV_TOKEN added
```

**Comment by willtebbutt** [comment]
Thanks for the assistance with this @maleadt -- strangely it doesn't appear to be the case that the result of running the buildkite CI is showing up with the other CI, and is no longer showing up as one of the CI runs at the bottom of this page. Any idea what might be going on?

(Thanks so much for your help with this. I really appreciate it!)

**Comment by maleadt** [comment]
> strangely it doesn't appear to be the case that the result of running the buildkite CI is showing up with the other CI, and is no longer showing up as one of the CI runs at the bottom of this page. Any idea what might be going on?

image

I assume that with the repository transfer, it's now part of an org that doesn't have the buildkite app yet. So you're back to step one in: https://github.com/JuliaGPU/buildkite?tab=readme-ov-file#steps-for-administrators

If you can make me temporarily an admin to the org, I can add the integration again.

**Comment by willtebbutt** [comment]
@yebai could you please make @maleadt an owner of the org briefly, so that this can be resolved?

**Comment by maleadt** [comment]
Thanks, you can remove me from the org.

The integration was set-up already, so I'm not really sure what's up here. I'll take another look, and if not I'll remove and re-create the pipeline if that's OK.

**Comment by maleadt** [comment]
Turns out the Buildkite integration was still tied to the previous org name, so had to be reinstalled. I've also re-created the pipeline; please push another commit, it should hopefully all work now.

**Comment by willtebbutt** [comment]
Hmmm it now seems to be the case that the tests are running, codecov is doing _something_ during buildkite's run, but the results don't seem to contribute the the coverage (it's still sitting at zero -- see [here](https://app.codecov.io/gh/compintell/Tapir.jl/pull/255?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=compintell).

@maleadt does everything look okay on your side of things now? I don't seem to be able to look at the complete CI logs on buildkite, so I'm not sure if there any some warnings I'm missing, or something like that.

**Comment by maleadt** [comment]
```
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_folder: Searching src for .jl files...
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/Tapir.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/Tapir.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/chain_rules_macro.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/chain_rules_macro.jl.3350233.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/chain_rules_macro.jl.3351887.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/codual.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/codual.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/fwds_rvs_data.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/fwds_rvs_data.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/interface.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/interface.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_folder: Searching src/interpreter for .jl files...
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/interpreter/abstract_interpretation.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/abstract_interpretation.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/abstract_interpretation.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/interpreter/bbcode.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/bbcode.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/interpreter/contexts.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/contexts.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/contexts.jl.3351887.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/contexts.jl.3352323.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/contexts.jl.3353543.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/contexts.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/interpreter/ir_normalisation.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/ir_normalisation.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/interpreter/ir_utils.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/ir_utils.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/interpreter/s2s_reverse_mode_ad.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/s2s_reverse_mode_ad.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/interpreter/zero_like_rdata.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/interpreter/zero_like_rdata.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_folder: Searching src/rrules for .jl files...
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/avoiding_non_differentiable_code.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/avoiding_non_differentiable_code.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/blas.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/rrules/blas.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/builtins.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/rrules/builtins.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/fastmath.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/fastmath.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/foreigncall.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/foreigncall.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/iddict.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/iddict.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/lapack.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/lapack.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/low_level_maths.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/low_level_maths.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/misc.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/misc.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/new.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/new.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/rrules/tasks.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/rrules/tasks.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/safe_mode.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/safe_mode.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/stack.jl
2024-09-16 13:46:55 CEST	┌ Info: CoverageTools.process_cov: Coverage file(s) for src/stack.jl do not exist.
2024-09-16 13:46:55 CEST	└ Assuming file has no coverage.
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/tangents.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/tangents.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/test_utils.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/test_utils.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_file: Detecting coverage for src/utils.jl
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/utils.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: CoverageTools.process_cov: processing src/utils.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/Tapir.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/chain_rules_macro.jl.3350233.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/chain_rules_macro.jl.3351887.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/codual.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/fwds_rvs_data.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/abstract_interpretation.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/abstract_interpretation.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/bbcode.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/contexts.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/contexts.jl.3351887.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/contexts.jl.3352323.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/contexts.jl.3353543.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/contexts.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/ir_normalisation.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/ir_utils.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/s2s_reverse_mode_ad.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/interpreter/zero_like_rdata.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/rrules/blas.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/rrules/builtins.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/safe_mode.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/tangents.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/test_utils.jl.3357948.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/utils.jl.3349178.cov
2024-09-16 13:46:55 CEST	[ Info: Removing src/utils.jl.3357948.cov
2024-09-16 13:46:56 CEST	[ Info: Submitting data to Codecov...
```

This does not report coverage for `ext/`, because the `dirs` argument defaults to only `src`: https://github.com/JuliaCI/julia-coverage-buildkite-plugin/blob/e16e9ebce8bdc3f485317938edbbc33dd5f17356/hooks/post-command#L7-L16

**Comment by willtebbutt** [comment]
@maleadt seemingly this still isn't working quite correctly -- the coverage for `ext/MooncakeCUDAExt.jl` is still down at 0% (other coverage changes are there because I've had to comment out some code to get CI to pass during the name change process).

Two things to note:
1. I've run coverage locally, and I'm seeing around 85% coverage of `ext/MooncakeCUDAExt.jl`, and
2. the package name has now changed to `Mooncake.jl`, but decryption of the secret seems to be working fine.

Is there anything else obvious that you can think of that might be causing the lack of coverage?

**Comment by maleadt** [comment]
Coverage info is being detected and submitted:

```
2024-09-19 12:24:34 CEST	[ Info: CoverageTools.process_file: Detecting coverage for ext/MooncakeCUDAExt.jl
2024-09-19 12:24:34 CEST	[ Info: CoverageTools.process_cov: processing ext/MooncakeCUDAExt.jl.1970102.cov
```

Not sure what the issue is.

**Comment by willtebbutt** [comment]
Thanks for taking a look.

The total number of files being uploaded to codecov is 10 -- see [here](https://app.codecov.io/gh/compintell/Mooncake.jl/commit/fc11c3cff590519d0664f4f4434253837acc966b) for the latest commit. Although it _looks_ like something has changed in the latest run, because it's now refusing to decrypt the secrets again (apologies for if the package rename messed this up).

Either way, I agree with you that it looks like somehow the coverage report is getting lost in transit between the buildkite runner and wherever codecov hosts the results.

**Comment by maleadt** [comment]
> Although it _looks_ like something has changed in the latest run, because it's now refusing to decrypt the secrets again (apologies for if the package rename messed this up).

Yes, it did. I'll probably have to recreate the pipeline again.

**Comment by willtebbutt** [comment]
@maleadt when you get a minute, would you mind recreating the pipeline? I'm keen to start tackling this again.

**Comment by maleadt** [comment]
I did that right after my previous comment, and Buildkite reveals that the Codecov secret is successfully decrypted. Or is there anything else missing?

**Comment by willtebbutt** [comment]
Hurrah! We have non-zero coverage from the buildkite run. I can now resolve the low coverage number, and then merge this. Thanks again for your assistance @maleadt

File: test/integration_testing/cuda.jl [hunk 3]
```diff
@@ -5,13 +5,13 @@ using CUDA
...
-        CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}(undef, 8, 8);
+        CuArray{Float32, 2, CUDA.DeviceMemory}(undef, 8, 8);
         interface_only=false,
     )
 
     # Check we can instantiate a CuArray.
     test_rule(
-        sr(123456), CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}, undef, 256;
+        sr(123456), CuArray{Float32, 1, CUDA.DeviceMemory}, undef, 256;
         interface_only=true, is_primitive=true,
```
Comment (willtebbutt) [inline test/integration_testing/cuda.jl]:
```suggestion
        interface_only=true, is_primitive=true, debug_mode=true,
```

**Pull request merged**